presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeometryUtils.java

```diff
@@ -0,0 +1,182 @@
+            return null;
+        }
+        else {
+            if (shapeBuffer.getInt(0) == Geometry.Type.Unknown.value()) {
+                return null;
+            }
+            else {
+                SpatialReference spatialReference = null;
+                if (wkid != WKID_UNKNOWN) {
+                    spatialReference = SpatialReference.create(wkid);
+                }
+                Geometry esriGeom = OperatorImportFromESRIShape.local().execute(0, Geometry.Type.Unknown, shapeBuffer);
+                OGCGeometry createdGeom = OGCGeometry.createFromEsriGeometry(esriGeom, spatialReference);
+                return createdGeom;
+            }
+        }
+    }
+
+    public static final OGCType getOGCType(int value)
+    {
+        if (value < 0 || value >= values.length) {
+            return OGCType.UNKNOWN;
+        }
+        else {
+            return values[value];
+        }
+    }
+
+    public static Slice serialize(OGCGeometry ogcGeometry)
+    {
+        int wkid;
+        try {
+            wkid = ogcGeometry.SRID();
+        }
+        catch (NullPointerException npe) {
+            wkid = 0;
+        }
+
+        OGCType ogcType;
+        String typeName;
+        try {
+            typeName = ogcGeometry.geometryType();
+
+            if (typeName.equals(POINT)) {
+                ogcType = OGCType.ST_POINT;
+            }
+            else if (typeName.equals(LINE_STRING)) {
+                ogcType = OGCType.ST_LINESTRING;
+            }
+            else if (typeName.equals(POLYGON)) {
+                ogcType = OGCType.ST_POLYGON;
+            }
+            else if (typeName.equals(MULTI_POINT)) {
+                ogcType = OGCType.ST_MULTIPOINT;
+            }
+            else if (typeName.equals(MULTI_LINE_STRING)) {
+                ogcType = OGCType.ST_MULTILINESTRING;
+            }
+            else if (typeName.equals(MULTI_POLYGON)) {
+                ogcType = OGCType.ST_MULTIPOLYGON;
+            }
+            else {
+                ogcType = OGCType.UNKNOWN;
+            }
+        }
+        catch (NullPointerException npe) {
+            ogcType = OGCType.UNKNOWN;
+        }
+
+        return serialize(ogcGeometry.getEsriGeometry(), wkid, ogcType);
+    }
+
+    private static Slice serialize(Geometry geometry, int wkid, OGCType type)
+    {
+        if (geometry == null) {
+            return null;
+        }
+
+        byte[] shape = GeometryEngine.geometryToEsriShape(geometry);
+
+        if (shape == null) {
+            return null;
+        }
+
+        Slice result = Slices.allocate(shape.length + SIZE_WKID + SIZE_TYPE);
+        result.setInt(0, wkid);
+        result.setByte(SIZE_WKID, (byte) type.getIndex());
+        result.setBytes(SIZE_WKID + SIZE_TYPE, shape);
+        return result;
+    }
+
+    public static OGCType getGeomType(Slice slice)
+    {
+        int s = (int) slice.getByte(SIZE_WKID);
+        return getOGCType(s);
+    }
+
+    public static int getWkid(Slice slice)
```

Comment:
This method doesn't seem to be used. Did you mean to make it private and call it from deserialize line 71?